client/landing/stepper/declarative-flow/internals/steps-repository/design-setup/site-setup-design-picker.tsx

```diff
@@ -255,7 +256,7 @@ const SiteSetupDesignPicker: Step = ( { navigation, flow } ) => {
 			params.append( 'redirect_to', window.location.href.replace( window.location.origin, '' ) );
 
 			window.location.href = `/checkout/${ encodeURIComponent(
-				siteSlugOrId
+				siteSlug || urlToSlug( site?.URL || '' )
```

Comment:
[nit] we might need to add a comment to describe that the `siteId` doesn't work for checkout and it might have a chance that the primary domain changes when checkout